src/groovy/heim/session/SessionService.groovy

```diff
@@ -127,7 +139,7 @@ class SessionService {
             throw new NoSuchResourceException(
                     "No such operational session: $sessionUUID")
         }
-
+        context.updateLastModified()
         SmartRSessionSpringScope.withActiveSession(
                 context, closure)
```

Comment:
This only updates the modification time when something _starts_ being done with the session. If what's being done is scheduling a task that takes a long time to run, then the session will immediately be deemed expired once the task finishes. It would be preferable to also update the modification time when a task finishes (but before it's marked as FINISHED/FAILED).